node-graph/graph-craft/src/expression.rs

```diff
@@ -0,0 +1,53 @@
+use crate::document::value::TaggedValue;
+use graphene_core::Node;
+use math_parser::evaluate;
+use math_parser::value::{Number, Value};
+
+/// A node that evaluates mathematical expressions during graph runtime
+#[derive(Debug)]
+pub struct ExpressionNode<T> {
+    expression: T,
+}
+
+impl<'i, T: Node<'i, ()>> Node<'i, ()> for ExpressionNode<T>
+where
+    T::Output: AsRef<str>,
+{
+    type Output = TaggedValue;
```

Comment:
Nodes should not return tagged value since then the output cannot be used by any other nodes.